frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/common.ts

```diff
@@ -34,13 +40,34 @@ export const initializeCommonMetadata = (
 
   entity.addLabel(`${TEMPLATE_OS_LABEL}/${osID}`, 'true');
   entity.addLabel(`${TEMPLATE_FLAVOR_LABEL}/${settings[VMSettingsField.FLAVOR]}`, 'true');
-  entity.addLabel(
-    `${TEMPLATE_WORKLOAD_LABEL}/${settings[VMSettingsField.WORKLOAD_PROFILE]}`,
-    'true',
-  );
 
-  entity.addLabel(LABEL_USED_TEMPLATE_NAME, getName(template));
-  entity.addLabel(LABEL_USED_TEMPLATE_NAMESPACE, getNamespace(template));
+  if (settings[VMSettingsField.WORKLOAD_PROFILE]) {
+    entity.addLabel(
+      `${TEMPLATE_WORKLOAD_LABEL}/${settings[VMSettingsField.WORKLOAD_PROFILE]}`,
+      'true',
+    );
+  }
+
+  if (template) {
+    entity.addLabel(LABEL_USED_TEMPLATE_NAME, getName(template));
+    entity.addLabel(LABEL_USED_TEMPLATE_NAMESPACE, getNamespace(template));
+  }
 
   return entity;
 };
+
+export const initializeCommonVMMetadata = (
+  { vmSettings }: CreateVMEnhancedParams,
+  entity: MutableVMWrapper | MutableVMTemplateWrapper,
+) => {
+  const settings = asSimpleSettings(vmSettings);
+  entity.addLabel(TEMPLATE_VM_NAME_LABEL, entity.getName()); // for pairing service-vm (like for RDP)
+
+  if (!entity.hasLabel(TEMPLATE_VM_DOMAIN_LABEL)) {
+    entity.addLabel(TEMPLATE_VM_DOMAIN_LABEL, entity.getName());
+  }
+
+  if (!entity.hasLabel(TEMPLATE_TYPE_APP)) {
+    entity.addLabel(TEMPLATE_TYPE_APP, settings[VMSettingsField.NAME]);
```

Comment:
why are there two approaches of getting the vmName? can we just use the settings one and make a variable out of it?